src/ngraph/pass/constant_folding.cpp

```diff
@@ -870,29 +870,33 @@ shared_ptr<op::Constant> fold_constant_binary(shared_ptr<op::Constant> a,
     }
     else
     {
-        if (std::dynamic_pointer_cast<op::Add>(binary))
+        if (auto add_node = std::dynamic_pointer_cast<op::Add>(binary))
```

Comment:
For CPU builders, can we instead check the auto-broadcast mode above instead ? assert it is not `BinaryElementwiseArithmetic` or `get_autob()` is None. I rather replace all the checks in the builders with just one check here.

Reply:
Ooh, great idea. Will do.

Reply:
@nmostafa Per this and our offline discussion, have moved the check into the `if` above and changed it from an assert to a fallback to reference. Note that the check is slightly different from what you describe: instead of checking `get_autob()` we check to make sure the input shapes match. This is because `ImplicitBroadcastElimination` does not actually change the autobroadcast spec, it just normalizes the shape so that downstream passes can safely ignore it.